packages/vega-parser/src/parsers/guides/axis-grid.js

```diff
@@ -38,49 +40,80 @@ export default function(spec, config, userEncode, dataRef, band) {
-    update[v2] = enter[v2] = {scale: vscale, range: 1, mult: sign, offset: offset};
+  if (!isSignal(orient)) {
+    if (orient === Top || orient === Bottom) {
+      u = 'x';
+      v = 'y';
+      s = 'height';
+    } else {
+      u = 'y';
+      v = 'x';
+      s = 'width';
+    }
+    v2 = v + '2';
+  
+    update[u] = enter[u] = exit[u] = tickPos;
+  
+    if (vscale) {
+      update[v] = enter[v] = {scale: vscale, range: 0, mult: sign, offset: offset};
+      update[v2] = enter[v2] = {scale: vscale, range: 1, mult: sign, offset: offset};
+    } else {
+      update[v] = enter[v] = {value: 0, offset: offset};
+      update[v2] = enter[v2] = {signal: s, mult: sign, offset: offset};
+    }
   } else {
-    update[v] = enter[v] = {value: 0, offset: offset};
-    update[v2] = enter[v2] = {signal: s, mult: sign, offset: offset};
+    update.x = enter.x = xAxisConditionalEncoding(orient.signal, tickPos, vscale ? { scale: vscale, range: 0, mult: sign, offset: offset} : { value: 0, offset: offset});
+    update.y = enter.y = xAxisConditionalEncoding(orient.signal, tickPos, vscale ? { scale: vscale, range: 0, mult: sign, offset: offset} : { value: 0, offset: offset}, false);
+    exit.x = xAxisConditionalEncoding(orient.signal, tickPos, null);
+    exit.y = xAxisConditionalEncoding(orient.signal, tickPos, null, false);
+    update.y2 = enter.y2 = xAxisConditionalEncoding(orient.signal, vscale ? {scale: vscale, range: 1, mult: sign, offset: offset} : {signal: xAxisExpr(orient.signal, 'height', 'width', false, false).signal, mult: sign, offset: offset}, null);
+    update.x2 = enter.x2 = xAxisConditionalEncoding(orient.signal, vscale ? {scale: vscale, range: 1, mult: sign, offset: offset} : {signal: xAxisExpr(orient.signal, 'height', 'width', false, false).signal, mult: sign, offset: offset}, null, false);
   }
-
+  
   return guideMark(RuleMark, AxisGridRole, null, Value, dataRef, encode, userEncode);
 }
 
 function offsetValue(offset, sign)  {
-  if (sign === 1) {
-    // do nothing!
-  } else if (!isObject(offset)) {
-    offset = sign * (offset || 0);
+  var entry;
+  if (!isSignal(sign)) {
+    if (sign === 1) {
+      // do nothing!
+    } else if (!isObject(offset)) {
+      offset = sign * (offset || 0);
+    } else {
+        entry = offset = extend({}, offset);
+  
+        while (entry.mult != null) {
+          if (!isObject(entry.mult)) {
+            entry.mult *= sign;
+            return offset;
+          } else {
+            entry = entry.mult = extend({}, entry.mult);
+          }
+        }
+    
+        entry.mult = sign;
+    }
+    return offset;
   } else {
-    var entry = offset = extend({}, offset);
-
-    while (entry.mult != null) {
-      if (!isObject(entry.mult)) {
-        entry.mult *= sign;
-        return offset;
-      } else {
-        entry = entry.mult = extend({}, entry.mult);
+    if (!isObject(offset)) {
+      return {
+        signal: `(${sign.signal}) === 1 ? 0 : (${sign.signal}) * (${offset || 0})`
```

Comment:
if `offset` is zero, we don't really need signal here.

Reply:
We need the else case though. Is there a workaround that I'm missing?

Reply:
It's not a big deal, I guess we can leave it as is.